proto/regen/ecocredit/v1/tx.proto

```diff
@@ -435,5 +439,61 @@ message MsgBridge {
   repeated Credits credits = 5;
 }
 
-// MsgBridgeResponse is the Msg/MsgBridge response type.
-message MsgBridgeResponse {}
\ No newline at end of file
+// MsgBridgeResponse is the Msg/Bridge response type.
+message MsgBridgeResponse {}
+
+// MsgBridgeReceive is the Msg/BridgeReceive request type.
+message MsgBridgeReceive {
+
+  // service_address is the account address of the service bridging the credits.
+  string service_address = 1;
+
+  // recipient is the recipient of the bridged credits.
+  string recipient = 2;
+
+  // amount is the amount of credits being bridged.
+  string amount = 3;
+
+  // A reference to a transaction or an event referencing the transaction
+  // which caused the transfer from another chain or registry.
+  // If not set (null), it means that the batch originate in Regen Ledger
+  // (doesn't come from external registry).
+  OriginTx origin_tx = 4;
+
+  // project_ref_id is the reference id of the project.
+  string project_ref_id = 5;
+
+  // project_jurisdiction is the project jurisdiction.
+  string project_jurisdiction = 6;
+
+  // start_date is the beginning of the period during which this credit batch
+  // was quantified and verified.
+  google.protobuf.Timestamp start_date = 7 [ (gogoproto.stdtime) = true ];
+
+  // end_date is the end of the period during which this credit batch was
+  // quantified and verified.
+  google.protobuf.Timestamp end_date = 8 [ (gogoproto.stdtime) = true ];
+
+  // project_metadata is the metadata for the project.
+  string project_metadata = 9;
+
+  // batch_metadata is the metadata for the batch.
+  string batch_metadata = 10;
+
+  // note is an arbitrary note to include in the origin tx.
+  string note = 11;
+
+  // class_id is the id of the class the credits being bridged belong to.
+  string class_id = 12;
```

Comment:
Should we create an embedded `Batch` and `Project` message to hold these values rather than prefixing each with either `project_` or `batch_` and creating one long list of mixed values? We could then including `origin_tx`, `note`, `start_date`, and `end_date` in `Batch` and `class_id` and `reference_id` in `Project`? It would also be nice to use the same field names so `ref_id` would be `reference_id`.

Reply:
embedded here https://github.com/regen-network/regen-ledger/pull/1174/commits/0510f6b0aa80e915d1a49a2e69baf7cc0bf35040